root/var/www/html/freepbx/rest/lib/libExtensions.php

```diff
@@ -121,6 +121,19 @@ function createExtension($mainextensionnumber,$delete=false){
         $stmt = $dbh->prepare($sql);
         $stmt->execute(array($extension));
 
+        // Use mainextension in displayname
+        if ($mainextensionnumber != $extension) {
+            // get current displayname
+            $sql = 'SELECT `data` FROM `sip` WHERE `id` = ? AND `keyword` = "callerid"';
+            $stmt = $dbh->prepare($sql);
+            $stmt->execute(array($extension));
+            $callerid = $stmt->fetchAll(\PDO::FETCH_ASSOC)[0]['data'];
+            $callerid = str_replace('<'.$extension.'>', '<'.$mainextensionnumber.'>',$callerid);
+            $sql = 'UPDATE IGNORE `sip` SET `data` = ? WHERE `id` = ? AND `keyword` = "callerid"';
+            $stmt = $dbh->prepare($sql);
+            $stmt->execute(array($callerid,$extension));
```

Comment:
Refer to https://mariadb.com/kb/en/replace-function/

```suggestion
            $sql = 'UPDATE IGNORE `sip` SET `data` = REPLACE(`data`, ?, ?) WHERE `id` = ? AND `keyword` = "callerid"';
            $stmt = $dbh->prepare($sql);
            $stmt->execute(array("<$extension>", "<$mainextensionnumber>", $extension));
```